factory/usr/lib/the-modeenv

```diff
@@ -5,10 +5,19 @@ cp /sysroot/run/image.fstab /run/
 # TODO:UC20: move this logic into snap-bootstrap to be incorporated with 
 # initramfs-mounts
 
+if grep -q snapd_recovery_mode=run /proc/cmdline; then
+    run_mode=1
+elif grep -q snapd_recovery_mode=install /proc/cmdline; then
+    if grep -q mode=run /run/mnt/ubuntu-data/system-data/var/lib/snapd/modeenv; then
+        run_mode=1
+    fi
+fi
+
```

Comment:
cmdline is more  complex. It could be the last argument. In 22 branch we have a shell script that properly parses and we can use that there.

I will propose something different tomorrow.

Reply:
do you mean `/usr/libexec/core/get`?
Yeah in 22 I used 
`run_mode="$(/usr/libexec/core/get-arg snapd_recovery_mode || true)"`
If we can backport that script, of course even better, then you can do
```
if [ -e "/run/mnt/data/system-data/var/lib/snapd/modeenv" ]; then
  run_mode="$(sed -n 's/mode=\([^[:space:]]*\)/\1/p' /run/mnt/data/system-data/var/lib/snapd/modeenv)"
else
  run_mode="$(/usr/libexec/core/get-arg snapd_recovery_mode || true)"
fi
```

Reply:
I have changed it. Take a look.